### ISSUE TITLE
Validation score reported even if no validation_df passed to CNN.train()

### ISSUE THREAD

**Issue by louisfh** [kitzeslab/opensoundscape#663]
I forgot to pass the validation_df to model.train.

```
from opensoundscape.torch.models.cnn import CNN
model = CNN('resnet34',classes=["1_class"],sample_duration=2.0)
model.train(train_df)
```
At the end of the training loop, after each epoch the model still prints out the MAP - and at the end states at which epoch the model with best 'validation score' appears:
```
Metrics:
	MAP: 0.827
....

Best Model Appears at Epoch 2 with Validation score 0.827.
```

**Comment by sammlapp** [comment]
By default, the training set is used if no validation set is provided. This allows you to get some sense of whether the model is training.

**Comment by sammlapp** [comment]
Maybe we should print a warning that the training data is being used as validation data?

**Comment by louisfh** [comment]
I think a warning that states that the 'validation' scores are based on training data would be a good fix. At the moment if users don't pass a valid_df, I think it's misleading that that we show them a metric and call it `Validation score` without any warning.
